### DISCUSSION TITLE
Bot is running but i can't give command to run the bot on server.

### DISCUSSION THREAD

**Discussion by NitishKumar-LD** [Rapptz/discord.py#9748] [Q&A]
import discord
from discord.ext import commands
intents = discord.Intents.all()
intents.members = True
intents.typing = True
intents.presences = True
intents.message_content = True
bot = commands.Bot(command_prefix='!', intents=intents)

events = {}

cooldown = commands.CooldownMapping.from_cooldown(1, 5, commands.BucketType.user)

@bot.event
async def on_ready():
    print(f'{bot.user} has connected to Discord!')

@bot.command(name='create_event', help='Create a new event')
async def create_event(ctx, date, time, description):
    # Add logic to store the event details (database, file, etc.)
    response = f'Event created on {date} at {time}. Description: {description}'
    await ctx.send(response)

@bot.command(name='list_events', help='List all upcoming events')
async def list_events(ctx):
    # Add logic to retrieve and display upcoming events
    events = ['Event 1', 'Event 2', 'Event 3']
    response = '\n'.join(events) if events else 'No upcoming events.'
    await ctx.send(response)

bot.run('token')

**Comment by MCausc78** [comment]
you're using text commands, (they are invoked through '[prefix]command <args...>'). To make slash commands, see: https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py
Also chatgpt is bad and doesnt know anything about discord.py, if you still want use text commands, then see https://github.com/Rapptz/discord.py/blob/master/examples/basic_bot.py

**Comment by AbstractUmbra** [comment]
This issue in unworkable to be honest. I can see you've provided code but you haven't really explained the issue?
Do you have errors in the console where you run the bot, or is there simply no output, no reaction or no response at all to the messages sent within discord?

I can see you set the prefix to `!`, does this mean you're attempting to invoke these commands like:-
- `!create_event date time description`

Or are you doing something else?

**Reply by NitishKumar-LD** [reply]
Actually the issue i get is if i  give command to the bot it doesn't work.

**Comment by felipeyx** [comment]
These are prefix/text commands, not slash commands:

@bot.command(name="create_event")

With:

bot = commands.Bot(command_prefix="!", ...)

the command must be invoked like:

!create_event 2024-03-10 18:00 "my event"

If that still does not trigger, make sure Message Content Intent is enabled in both places:

intents.message_content = True

and in Discord Developer Portal → Bot → Privileged Gateway Intents → Message Content Intent.

If what you actually want is /create_event, use app_commands.command or commands.hybrid_command and sync the command tree.
